### ISSUE TITLE
Implement save/load packets.

### ISSUE THREAD

**Issue by Andoryuuta** [ErupeServer/Erupe#4]
Persistent:
- [x] MsgMhfSavedata/MsgMhfLoaddata -- Main character save data
- [ ] MsgMhfSaveScenarioData/MsgMhfLoadScenarioData -- Scenario data
- [x] MsgMhfSavePlateData/MsgMhfLoadPlateData -- Sigil related?
- [x] MsgMhfSavePlateBox/MsgMhfLoadPlateBox -- G Rank sigils which are inserted into weapons to give buffs
- [x] MsgMhfSavePartner/MsgMhfLoadPartner -- Human partner
- [x] MsgMhfSaveOtomoAirou/MsgMhfLoadOtomoAirou -- Cat partner
- [x] MsgMhfSaveDecoMyset/MsgMhfLoadDecoMyset -- Decorations associated with your saved armor sets
- [x] MsgMhfSaveHunterNavi/MsgMhfLoadHunterNavi -- Hunter navigation
- [x] MsgMhfSavePlateMyset/MsgMhfLoadPlateMyset -- "Presumably the sigils equivalent for the decorations version above"
- [x] MsgMhfSaveRengokuData/MsgMhfLoadRengokuData -- Hunter's road

Non-persistent:
- [x] MsgSysSetUserBinary/MsgSysGetUserBinary -> Non-persistent user binary pieces for looks, etc.
- [x] MsgSysSetStageBinary/MsgSysGetStageBinary -> Non-persistent stage data set by the stage creator/owner.

Unknown:
- [ ] MsgSysSetObjectBinary/MsgSysGetObjectBinary -> unused?

**Comment by SirFist** [comment]
Couple notes from looking at it beyond a cursory glance.

PlateBox is indeed the sigil box and with values from a decompressed save seems to be pretty straight forward:
```
Swaxe up +12 / Elemental +11/ Elemental +8
00 6F 00 02 00 02 00 0C 00 0B 00 08 00 00 00 00 
Bow Up +15 / Attack +11/ Attack +5
00 6D 00 01 00 01 00 0F 00 0B 00 05 00 00 00 00 
```
16 bytes
3x 2 bytes for roll types
3x 2 bytes for roll values

All 00s for empty slot, truncated down to last 'empty' sigil and terminated after last valid one with 
```00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00```
presumably if you have a valid sigil in the 300th slot it terminates with an actual sigil instead.

PlateMyset is just the same format from individual sigils above grouped into three for each preset with empty ones indicating no sigils resulting in 48 bytes per 'slot', e.g. :
```
Preset 1:
Sns Up +15 / Status / SnS Beam Slash
Attack +15 / Attack +15 / Attack +8
Attack +13 / Element +15 / Element +7
00 67 00 03 00 3F 00 0F 00 01 00 01 00 00 00 00 
00 01 00 01 00 01 00 0F 00 0F 00 08 00 00 00 00 
00 01 00 02 00 02 00 0D 00 0F 00 07 00 00 00 00
Preset 2:
DS Up +15 / Elemental +10 / Status
Attack +4 / Attack +12 / Attack +14
Attack +15 / Attack +15 / Attack +8
00 69 00 02 00 03 00 0F 00 0A 00 01 00 00 00 00 
00 01 00 01 00 01 00 04 00 0C 00 0E 00 00 00 00
00 01 00 01 00 01 00 0F 00 0F 00 08 00 00 00 00
```

PlateData is looks to be all the things that apply 'on top' of equipment that isn't in the standard save, so the transmog system for armours and the sigil system for weapons.
Sigils wise it's identical to the above two and has 3 values just like the MySet with each slot taking up 48 bytes.

Transmogs look simply have the replacement visual ID set on them at the 12th byte.
```
00 00 00 00 00 00 00 00 00 00 00 00 2E 60 00 00
00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00
00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00
```
The only save file I have for these ends with a transmog, and trims after the first 16 bytes 
```
00 00 00 00 00 00 00 00 00 00 00 00 31 F2 00 00
```
Armour IDs are apparently big endian.

Not directly related, but the actual transmog save for what you have unlocked is the **MSG_MHF_GET_EQUIP_SKIN_HIST** bitmask response and isn't tied to the box or plate stuff at all but is another save that needs to be made persistent and one that is only ever server sided.

**Comment by Andoryuuta** [comment]
Just to add historical context to this issue thread, almost all of these were implemented in SirFist's PR (#9).

The following packets still need research done to determine their use and if they need to be persistent or temporary in memory:
MsgMhfSaveScenarioData/MsgMhfLoadScenarioData
MsgSysSetObjectBinary/MsgSysGetObjectBinary
